src/Components/Components/src/Properties/ILLink.Descriptors.xml

```diff
@@ -0,0 +1,11 @@
+<linker>
```

Comment:
It would be good to add explanation why these things are unconditionally rooted (e.g. if the type is referenced in any way - a local variable, a never assigned field of this type, etc. it's going to be preserved in its entirety. Usually it's not desirable to have any ILLink.Descriptors.xml and if there are, there should be a good record why.

Reply:
I've reduced this list to only include Blazor components. In general, the way these get is not friendly with linker analysis and there isn't an alternate place to put these attributes. https://github.com/mono/linker/issues/1806 would help remove these.

Reply:
Will we need to remember to add all new built-in components in the future to this file? What happens if we don't? Does it just fail to make that new component type eligible for being stripped out?

Reply:
Any unsuppressed linker warning causes a local build to produce a warning when it builds the newly added test project and the CI build to fail because it has WarningsAsErrors=true.